### PR TITLE
feat: Add TensorShape class to the TensorFlow frontend

### PR DESCRIPTION
**Pull request by Sam-Armstrong**
I still need to make some more changes here and add the rest of the tests, but just checking if you think the test I've added looks ok @AnnaTz? And how could I improve it? Thanks!

### CODE REVIEW

File: ivy/functional/frontends/tensorflow/tensor.py [hunk 1]
```diff
@@ -228,6 +228,70 @@ def __iter__(self):
...
 
 
+class TensorShape:
+    # TODO: there are still some methods that may need implementing
+
+    def __init__(self, dims):
+        self._dims = tuple(dims)
+
+    def __repr__(self):
+        return str(self._dims)
```
Comment (AnnaTz) [inline ivy/functional/frontends/tensorflow/tensor.py:238]: Is this really the original implementation? I think we should be printing something like this 
image

**Review by AnnaTz — CHANGES_REQUESTED** [review]

File: ivy/functional/frontends/tensorflow/tensor.py (hunk 1 above)
Comment (Sam-Armstrong) [reply ivy/functional/frontends/tensorflow/tensor.py:238]: Yeah you're right, just looked at the source code. For some reason when I ran it in Python locally it returned the shape in the tuple format which is why I did it like this, not sure what was causing that

**Comment by Sam-Armstrong** [comment]
Strangely there doesn't seem to be a property of method for directly accessing the value of `_dims`, other than one which is deprecated so probably not worth including https://github.com/tensorflow/tensorflow/blob/v2.14.0/tensorflow/python/framework/tensor_shape.py

**Comment by AnnaTz** [comment]
> Strangely there doesn't seem to be a property of method for directly accessing the value of `_dims`, other than one which is deprecated so probably not worth including https://github.com/tensorflow/tensorflow/blob/v2.14.0/tensorflow/python/framework/tensor_shape.py

I'm guessing we will need it at some point but no reason to add it until we do.

File: ivy/functional/frontends/tensorflow/tensor.py [hunk 2]
```diff
@@ -228,6 +228,115 @@ def __iter__(self):
             yield self[i]
 
 
+class TensorShape:
+    # TODO: there are still some methods that may need implementing
+
+    def __init__(self, dims):
+        self._ivy_shape = dims if isinstance(dims, ivy.Shape) else ivy.shape(dims)
+        self._dims = self._ivy_shape._shape
```
Comment (AnnaTz) [inline ivy/functional/frontends/tensorflow/tensor.py:236]: I would suggest keeping only `_dims` here and inferring the ivy shape in the corresponding property.
Other than that, looks good!

**Review by AnnaTz — CHANGES_REQUESTED** [review]

File: ivy/functional/frontends/tensorflow/tensor.py (hunk 2 above)
Comment (Sam-Armstrong) [reply ivy/functional/frontends/tensorflow/tensor.py]: Sure, thanks Anna!

**Pull request merged**